bundlewrap/items/files.py

```diff
@@ -133,9 +138,74 @@ def content_processor_text(item):
     'jinja2': content_processor_jinja2,
     'mako': content_processor_mako,
     'text': content_processor_text,
+    'download': None,
 }
 
 
+def download_file(item):
+    file_name_hashed = md5(item.attributes['source'].encode('UTF-8')).hexdigest()
+
+    cache_path = getenv('BW_FILE_DOWNLOAD_CACHE')
+    if cache_path:
+        remove_dir = None
+        file_path = join(cache_path, file_name_hashed)
+        lock_dir = join(cache_path, '{}.bw_lock'.format(file_name_hashed))
+
+        makedirs(cache_path, exist_ok=True)
+    else:
+        remove_dir = join(gettempdir(), 'bw-file-download-cache-{}'.format(getpid()))
+        file_path = join(remove_dir, file_name_hashed)
+        lock_dir = join(remove_dir, '{}.bw_lock'.format(file_name_hashed))
+
+        makedirs(remove_dir, exist_ok=True)
+
+    io.debug(_('lock dir for {} is {}'.format(item.name, lock_dir)))
+
+    while True:
```

Comment:
should probably wrap this in `with io.job(_("waiting for download [...]` (and the actual download as well of course)